packages/financial-templates-lib/src/clients/OptimisticOracleClient.ts

```diff
@@ -286,15 +287,31 @@ export class OptimisticOracleClient {
               )
               .call();
 
+            let timestampForDvmRequest = disputeEvent.returnValues.timestamp;
+            if (this.oracleType === OptimisticOracleType.OptimisticOracleV2) {
+              const request = await (this.oracle as OptimisticOracleV2Web3).methods
+                .getRequest(
+                  disputeEvent.returnValues.requester,
+                  disputeEvent.returnValues.identifier,
+                  disputeEvent.returnValues.timestamp,
+                  disputeEvent.returnValues.ancillaryData
+                )
+                .call({ from: this.oracle.options.address });
+
+              // Check if the request is an event based request
+              if (request[4][0]) {
```

Comment:
Can we not unpack this struct by name rather than array element? Does web3 not decode it correctly?

Reply:
@mrice32, it's an issue with the contract types, they don't have the requestSettings props completely mapped. I've added a workaround in[ this last commit](https://github.com/UMAprotocol/protocol/pull/4567/commits/2074f8f19a040f322ce5a9db7480773d85dc58b1). It's not the cleanest but don't see any simple solution.